release-21.1: workload/movr: add --multi-region and --survive settings

otan opened the pull request:
Backport 1/1 commits from #65493.

/cc @cockroachdb/release

---

Release note (general change): Added multi-region workloads for
`cockroach demo movr --geo-partitioned-replicas`. Specifying
`--multi-region` allows for multi-region workloads, and specifying
`--survive` allows for surviving AZ or REGION failures.

otan committed:
workload/movr: add --multi-region and --survive settings

Release note (general change): Added multi-region workloads for
`cockroach demo movr --geo-partitioned-replicas. Specifying
`--multi-region` allows for multi-region workloads, and specifying
`--survive` allows for surviving AZ or REGION failures. Specifying
`--infer-crdb-region-column` also infers the crdb_region for
REGIONAL BY ROW tables.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/65642)
<!-- Reviewable:end -->

ajstorm reviewed (commented):
I feel uncomfortable backporting this change until we have at least _some_ testing around it.  Can we address that before we proceed on backporting? 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/65642#-:-MaYkZTn7Xvx4hUHAwMd:b1ii0rp)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
the existing tests for these heavily flake on CI, probably because the machine is hammered. i will not have time to investigate this (we all remember what happened last time i dove into a cli issue ;P)

See also: https://github.com/cockroachdb/cockroach/issues/50970, https://github.com/cockroachdb/cockroach/blob/master/pkg/cli/interactive_tests/test_demo_partitioning.tcl.disabled.

```
.200706 05:17:36.196939091 EXPECT TEST: START TEST: Expect partitioning succeeds
.200706 05:18:36.743634193 EXPECT TEST: TIMEOUT WAITING FOR "movr>"

spawn /bin/bash
PS1=':''/# '
roach@7059f4574d07:~$ PS1=':''/# '
:/# 
.200706 05:17:36.196939091 EXPECT TEST: START TEST: Expect partitioning succeeds
/cockroach/cockroach demo --geo-partitioned-replicas
#
# --geo-partitioned replicas operates on a 9 node cluster.
# The cluster size has been changed from the default to 9 nodes.
#
# Welcome to the CockroachDB demo database!
#
# You are connected to a temporary, in-memory CockroachDB cluster of 9 nodes.
#
# This demo session will attempt to enable enterprise features
# by acquiring a temporary license from Cockroach Labs in the background.
# To disable this behavior, set the environment variable
# COCKROACH_SKIP_ENABLING_DIAGNOSTIC_REPORTING=true.
#
# Beginning initialization of the movr dataset, please wait...

.200706 05:18:36.743634193 EXPECT TEST: TIMEOUT WAITING FOR "movr>"
non-zero exit code: 1
```

otan committed:
cli/interactive_tests: re-enable test_demo_partitioning.tcl

This seemed to always time out in earlier runs, so give it a slightly
beefier timeout.

Release note: None

otan commented:
let's see if https://github.com/cockroachdb/cockroach/pull/65687 deflakes partitioning, if so i have a path forward to adding the tests without flaking.

ajstorm reviewed (commented):
SGTM.  I'll stay tuned.  Poke again when you want me to have another look.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/65642#-:-Macu-LD6nvlmmx-xEB5:b-ynkfzw)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan committed:
cli/interactive_tests: add tests for --multi-region on demo movr

Release note: None

otan commented:
alright, the tests are backported too

ajstorm reviewed (approved):
Nice.  <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/65642#-:-Mb1GU616hVmpN-16AO6:big2ka0)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)



<!-- Sent from Reviewable.io -->
